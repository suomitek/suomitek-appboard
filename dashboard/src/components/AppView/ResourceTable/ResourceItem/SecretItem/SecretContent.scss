.SecretContainer {
  margin: 0;
  padding: 0;
  margin-left: 0.2em;
  background: inherit;
}

.SecretContent {
  margin: 0;
  min-width: 0;
  white-space: pre-wrap;
  overflow-wrap: break-word;
  padding-left: 0.3em;
  padding-right: 0.3em;
  color: inherit;
}

.SecretType {
  overflow-wrap: break-word;
}

.secretDatum {
  background-color: transparent;
  border: none;
  cursor: pointer;
  text-decoration: underline;
  display: inline;
  margin: 0;
  padding: 0;
  outline: none;
  color: #0d6cd4;
}

.secretDatum:hover {
  color: #dd751d;
  text-decoration: none;
}
