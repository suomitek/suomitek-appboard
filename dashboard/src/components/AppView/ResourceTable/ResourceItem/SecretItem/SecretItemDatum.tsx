import * as React from "react";
import { Eye, EyeOff } from "react-feather";

interface ISecretItemDatumProps {
  name: string;
  value: string;
}

interface ISecretItemDatumState {
  hidden: boolean;
}

class SecretItemDatum extends React.PureComponent<ISecretItemDatumProps, ISecretItemDatumState> {
  // Secret datum is hidden by default
  public state: ISecretItemDatumState = {
    hidden: true,
  };

  public render() {
    const { name, value } = this.props;
    const { hidden } = this.state;
    const decodedValue = atob(value);
    return (
      <span className="flex">
        <button className="secretDatum" onClick={this.toggleDisplay}>
          {hidden ? <Eye /> : <EyeOff />}
        </button>
        <span className="flex margin-l-normal">
          <span>{name}:</span>
          {hidden ? (
            <span className="margin-l-small">{decodedValue.length} bytes</span>
          ) : (
            <pre className="SecretContainer">
              <code className="SecretContent">{decodedValue}</code>
            </pre>
          )}
        </span>
      </span>
    );
  }

  private toggleDisplay = () => {
    this.setState({
      hidden: !this.state.hidden,
    });
  };
}

export default SecretItemDatum;
