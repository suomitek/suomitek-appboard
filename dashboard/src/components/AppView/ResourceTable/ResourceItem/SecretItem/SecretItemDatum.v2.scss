.secret-datum-text {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  height: 100%;
  padding-left: 0.3rem;
}

.secret-datum-icon {
  background: transparent;
  border: none;
  cursor: pointer;
  cds-icon {
    --color: var(--clr-link-color, #0072a3);
    &:hover {
      --color: var(--clr-link-active-color, #9e57bc);
    }
  }
  &:focus {
    outline: none;
  }
}

.secret-datum-content {
  border-bottom: 1px solid var(--clr-forms-border-color, #b3b3b3) !important;
}
