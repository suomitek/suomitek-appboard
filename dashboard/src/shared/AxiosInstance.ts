import Axios, { AxiosError, AxiosInstance, AxiosRequestConfig, AxiosResponse } from "axios";

import { Action, Store } from "redux";
import { ThunkDispatch } from "redux-thunk";
import actions from "../actions";
import { Auth } from "./Auth";
import {
  ConflictError,
  ForbiddenError,
  InternalServerError,
  IStoreState,
  NotFoundError,
  UnauthorizedError,
  UnprocessableEntity,
} from "./types";

export function addAuthHeaders(axiosInstance: AxiosInstance) {
  axiosInstance.interceptors.request.use((config: AxiosRequestConfig) => {
    const authToken = Auth.getAuthToken();
    if (authToken) {
      config.headers.Authorization = `Bearer ${authToken}`;
    }
    return config;
  });
}

export function addErrorHandling(axiosInstance: AxiosInstance, store: Store<IStoreState>) {
  axiosInstance.interceptors.response.use(
    response => response,
    e => {
      const dispatch = store.dispatch as ThunkDispatch<IStoreState, null, Action>;
      const err: AxiosError = e;
      if (
        err.code === undefined &&
        err.message === "Network Error" &&
        !err.response &&
        Auth.usingOIDCToken()
      ) {
        // The OIDC token is no longer valid, logout
        dispatch(actions.auth.expireSession());
      }
      let message = err.message;
      if (err.response) {
        if (err.response.data.message) {
          message = err.response.data.message;
        }
        if (typeof err.response.data === "string") {
          message = err.response.data;
        }
      }

      const dispatchErrorAndLogout = (m: string) => {
        // Global action dispatch to log the user out
        dispatch(actions.auth.authenticationError(m));
        // Expire the session if we are using OIDC tokens and
        // logout either way.
        dispatch(actions.auth.expireSession());
      };
      const response = err.response as AxiosResponse;
      switch (response && response.status) {
        case 401:
          dispatchErrorAndLogout(message);
          return Promise.reject(new UnauthorizedError(message));
        case 403:
          // A 403 directly from the auth proxy requires reauthentication.
          if (Auth.usingOIDCToken() && Auth.is403FromAuthProxy(response)) {
            dispatchErrorAndLogout(message);
          }
          return Promise.reject(new ForbiddenError(message));
        case 404:
          return Promise.reject(new NotFoundError(message));
        case 409:
          return Promise.reject(new ConflictError(message));
        case 422:
          return Promise.reject(new UnprocessableEntity(message));
        case 500:
          return Promise.reject(new InternalServerError(message));
        default:
          return Promise.reject(new Error(message));
      }
    },
  );
}

// Error handling is added with an interceptor in index.tsx
export const axios = Axios.create();
// Authorization headers and error handling are added with an interceptor in index.tsx
export const axiosWithAuth = Axios.create();
