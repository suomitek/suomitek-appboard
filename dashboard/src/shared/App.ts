import * as url from "shared/url";
import { axiosWithAuth } from "./AxiosInstance";
import { hapi } from "./hapi/release";
import { IAppOverview, IChartVersion } from "./types";

export const KUBEOPS_ROOT_URL = "api/tiller-deploy/v1";

export class App {
  public static async create(
    cluster: string,
    namespace: string,
    releaseName: string,
    chartNamespace: string,
    chartVersion: IChartVersion,
    values?: string,
  ) {
    const chartAttrs = chartVersion.relationships.chart.data;
    const endpoint = url.kubeops.releases.list(cluster, namespace);
    const { data } = await axiosWithAuth.post(endpoint, {
      appRepositoryResourceName: chartAttrs.repo.name,
      appRepositoryResourceNamespace: chartNamespace,
      chartName: chartAttrs.name,
      releaseName,
      values,
      version: chartVersion.attributes.version,
    });
    return data;
  }

  public static async upgrade(
    cluster: string,
    namespace: string,
    releaseName: string,
    chartNamespace: string,
    chartVersion: IChartVersion,
    values?: string,
  ) {
    const chartAttrs = chartVersion.relationships.chart.data;
    const endpoint = url.kubeops.releases.get(cluster, namespace, releaseName);
    const { data } = await axiosWithAuth.put(endpoint, {
      appRepositoryResourceName: chartAttrs.repo.name,
      appRepositoryResourceNamespace: chartNamespace,
      chartName: chartAttrs.name,
      releaseName,
      values,
      version: chartVersion.attributes.version,
    });
    return data;
  }

  public static async rollback(
    cluster: string,
    namespace: string,
    releaseName: string,
    revision: number,
  ) {
    const endpoint = url.kubeops.releases.get(cluster, namespace, releaseName);
    const { data } = await axiosWithAuth.put(
      endpoint,
      {},
      {
        params: {
          action: "rollback",
          revision,
        },
      },
    );
    return data;
  }

  public static async delete(
    cluster: string,
    namespace: string,
    releaseName: string,
    purge: boolean,
  ) {
    let endpoint = url.kubeops.releases.get(cluster, namespace, releaseName);
    if (purge) {
      endpoint += "?purge=true";
    }
    const { data } = await axiosWithAuth.delete(endpoint);
    return data;
  }

  public static async listApps(cluster: string, namespace?: string, allStatuses?: boolean) {
    let endpoint = namespace
      ? url.kubeops.releases.list(cluster, namespace)
      : url.kubeops.releases.listAll(cluster);
    if (allStatuses) {
      endpoint += "?statuses=all";
    }
    const { data } = await axiosWithAuth.get<{ data: IAppOverview[] }>(endpoint);
    return data.data;
  }

  public static async getRelease(cluster: string, namespace: string, name: string) {
    const { data } = await axiosWithAuth.get<{ data: hapi.release.Release }>(
      url.kubeops.releases.get(cluster, namespace, name),
    );
    return data.data;
  }
}
