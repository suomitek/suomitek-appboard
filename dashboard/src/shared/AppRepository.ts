import { axiosWithAuth } from "./AxiosInstance";
import { APIBase } from "./Kube";
import { ICreateAppRepositoryResponse } from "./types";
import * as url from "./url";

export class AppRepository {
  public static async list(namespace: string) {
    const { data } = await axiosWithAuth.get(AppRepository.getSelfLink(namespace));
    return data;
  }

  public static async get(name: string, namespace: string) {
    const { data } = await axiosWithAuth.get(AppRepository.getSelfLink(namespace, name));
    return data;
  }

  public static async resync(name: string, namespace: string) {
    const repo = await AppRepository.get(name, namespace);
    repo.spec.resyncRequests = repo.spec.resyncRequests || 0;
    repo.spec.resyncRequests++;
    const { data } = await axiosWithAuth.put(AppRepository.getSelfLink(namespace, name), repo);
    return data;
  }

  public static async update(
    name: string,
    namespace: string,
    repoURL: string,
    authHeader: string,
    customCA: string,
    syncJobPodTemplate: any,
    registrySecrets: string[],
  ) {
    const { data } = await axiosWithAuth.put<ICreateAppRepositoryResponse>(
      url.backend.apprepositories.update(namespace, name),
      {
        appRepository: { name, repoURL, authHeader, customCA, syncJobPodTemplate, registrySecrets },
      },
    );
    return data;
  }

  public static async delete(name: string, namespace: string) {
    const { data } = await axiosWithAuth.delete(
      url.backend.apprepositories.delete(name, namespace),
    );
    return data;
  }

  // create uses the kubeapps backend API
  // TODO(mnelson) Update other endpoints to similarly use the backend API, removing the need
  // for direct k8s api access (for this resource, at least).
  public static async create(
    name: string,
    namespace: string,
    repoURL: string,
    authHeader: string,
    customCA: string,
    syncJobPodTemplate: any,
    registrySecrets: string[],
  ) {
    const { data } = await axiosWithAuth.post<ICreateAppRepositoryResponse>(
      url.backend.apprepositories.create(namespace),
      {
        appRepository: { name, repoURL, authHeader, customCA, syncJobPodTemplate, registrySecrets },
      },
    );
    return data;
  }

  public static async validate(repoURL: string, authHeader: string, customCA: string) {
    const { data } = await axiosWithAuth.post<any>(url.backend.apprepositories.validate(), {
      appRepository: { repoURL, authHeader, customCA },
    });
    return data;
  }

  private static APIBase: string = APIBase;
  private static APIEndpoint: string = `${AppRepository.APIBase}/apis/kubeapps.com/v1alpha1`;
  private static getSelfLink(namespace: string, name?: string): string {
    return `${AppRepository.APIEndpoint}/namespaces/${namespace}/apprepositories${
      name ? `/${name}` : ""
    }`;
  }
}
