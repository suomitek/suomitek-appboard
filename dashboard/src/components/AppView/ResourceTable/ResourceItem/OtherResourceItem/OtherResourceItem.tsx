import * as React from "react";
import ResourceRef from "../../../../../shared/ResourceRef";
import { IResource } from "../../../../../shared/types";

export interface IOtherResourceProps {
  resource: IResource | ResourceRef;
}

export const OtherResourceColumns: React.SFC = () => {
  return (
    <React.Fragment>
      <th className="col-6">NAME</th>
      <th className="col-6">KIND</th>
    </React.Fragment>
  );
};

const OtherResourceItem: React.SFC<IOtherResourceProps> = props => {
  const { resource } = props;
  let name;
  const fullResource = resource as IResource;
  if (fullResource.metadata && fullResource.metadata.name) {
    name = fullResource.metadata.name;
  } else {
    const resourceRef = resource as ResourceRef;
    name = resourceRef.name;
  }
  return (
    <>
      <td className="col-6">{name}</td>
      <td className="col-6">{resource.kind}</td>
    </>
  );
};

export default OtherResourceItem;
