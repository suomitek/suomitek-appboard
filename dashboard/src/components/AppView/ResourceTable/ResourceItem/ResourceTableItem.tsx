import * as React from "react";
import { AlertTriangle } from "react-feather";

import { IK8sList, IKubeItem, IResource, ISecret } from "../../../../shared/types";
import LoadingWrapper, { LoaderType } from "../../../LoadingWrapper";
import DaemonSetItemRow from "./DaemonSetItem";
import DeploymentItemRow from "./DeploymentItem";
import OtherResourceItem from "./OtherResourceItem";
import SecretItem from "./SecretItem/SecretItem";
import ServiceItem from "./ServiceItem/ServiceItem";
import StatefulSetItemRow from "./StatefulSetItem";

interface IResourceItemProps {
  name: string;
  resource?: IKubeItem<IResource | ISecret | IK8sList<IResource | ISecret, {}>>;
  watchResource: () => void;
  closeWatch: () => void;
  avoidEmptyResouce?: boolean;
}

class WorkloadItem extends React.Component<IResourceItemProps> {
  public componentDidMount() {
    this.props.watchResource();
  }

  public componentWillUnmount() {
    this.props.closeWatch();
  }

  public render() {
    const { resource } = this.props;
    return this.renderInfo(resource);
  }

  private renderInfo(
    resource?: IKubeItem<IResource | ISecret | IK8sList<IResource | ISecret, {}>>,
  ) {
    const { name } = this.props;
    if (resource === undefined || resource.isFetching) {
      return (
        <tr className="flex">
          <td className="col-3">{name}</td>
          <td className="col-9">
            <LoadingWrapper type={LoaderType.Placeholder} />
          </td>
        </tr>
      );
    }
    if (resource.error) {
      return (
        <tr className="flex">
          {" "}
          <td className="col-3">{name}</td>
          <td className="col-9">
            <span className="flex">
              <AlertTriangle />
              <span className="flex margin-l-normal">Error: {resource.error.message}</span>
            </span>
          </td>
        </tr>
      );
    }
    if (resource.item) {
      const listItem = resource.item as IK8sList<IResource | ISecret, {}>;
      if (listItem.items) {
        if (listItem.items.length === 0) {
          if (this.props.avoidEmptyResouce) {
            return null;
          }
          return (
            <tr className="flex">
              <td className="col-12">No resource found</td>
            </tr>
          );
        }
        return listItem.items.map(i => (
          <tr key={i.metadata.selfLink} className="flex">
            {this.renderResource(i)}
          </tr>
        ));
      }
      return <tr className="flex">{this.renderResource(resource.item as IResource)}</tr>;
    }
    return <span>No resource found</span>;
  }

  private renderResource = (r: IResource | ISecret) => {
    const plainResource = r as IResource;
    // The resource kind may not be available for Lists
    // so we need to infer it from the selfLink
    const parsedLink = r.metadata.selfLink.split("/");
    if (parsedLink.length < 2) {
      // Unknown selflink
      return "";
    }
    // For a single resource, the type is the second-to-last item
    // e.g. /api/v1/namespaces/default/pods/foo
    const type = parsedLink[parsedLink.length - 2];
    switch (type) {
      case "deployments":
        return <DeploymentItemRow resource={plainResource} />;
      case "statefulsets":
        return <StatefulSetItemRow resource={plainResource} />;
      case "daemonsets":
        return <DaemonSetItemRow resource={plainResource} />;
      case "services":
        return <ServiceItem resource={plainResource} />;
      case "secrets":
        return <SecretItem resource={r as ISecret} />;
      default:
        return <OtherResourceItem resource={plainResource} />;
    }
  };
}

export default WorkloadItem;
