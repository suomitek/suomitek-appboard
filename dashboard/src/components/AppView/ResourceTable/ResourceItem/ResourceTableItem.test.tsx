import { shallow } from "enzyme";
import context from "jest-plugin-context";
import * as React from "react";

import itBehavesLike from "../../../../shared/specs";
import { IKubeItem, IResource } from "../../../../shared/types";
import DeploymentItemRow from "./DeploymentItem/DeploymentItem";
import OtherResourceItem from "./OtherResourceItem";
import ResourceTableItem from "./ResourceTableItem";
import SecretItem from "./SecretItem";

const kubeItem: IKubeItem<IResource> = {
  isFetching: false,
};

const defaultProps = {
  name: "foo",
  watchResource: jest.fn(),
  closeWatch: jest.fn(),
  statusFields: [],
};

describe("componentDidMount", () => {
  it("calls watchResource", () => {
    const mock = jest.fn();
    shallow(<ResourceTableItem {...defaultProps} watchResource={mock} />);
    expect(mock).toHaveBeenCalled();
  });
});

describe("componentWillUnmount", () => {
  it("calls closeWatch", () => {
    const mock = jest.fn();
    const wrapper = shallow(<ResourceTableItem {...defaultProps} closeWatch={mock} />);
    wrapper.unmount();
    expect(mock).toHaveBeenCalled();
  });
});

context("when fetching resources", () => {
  [undefined, { isFetching: true }].forEach(resource => {
    itBehavesLike("aLoadingComponent", {
      component: ResourceTableItem,
      props: {
        ...defaultProps,
        resource,
      },
    });

    it("displays the name of the resource", () => {
      const wrapper = shallow(<ResourceTableItem {...defaultProps} resource={resource} />);
      expect(wrapper.text()).toContain("foo");
    });
  });
});

context("when there is an error fetching the resource", () => {
  const resource = {
    error: new Error('deployments "foo" not found'),
    isFetching: false,
  };
  const wrapper = shallow(<ResourceTableItem {...defaultProps} resource={resource} />);

  it("diplays the resource name in the first column", () => {
    expect(
      wrapper
        .find("td")
        .first()
        .text(),
    ).toEqual("foo");
  });

  it("displays the error message in the second column", () => {
    expect(
      wrapper
        .find("td")
        .at(1)
        .text(),
    ).toContain('Error: deployments "foo" not found');
  });
});

context("when there is a valid resouce", () => {
  it("renders info about the resource status", () => {
    const deployment = {
      metadata: {
        name: "foo",
        selfLink: "/deployments/foo",
      },
      status: { replicas: 1, updatedReplicas: 1, availableReplicas: 1 },
    } as IResource;
    kubeItem.item = deployment;
    const wrapper = shallow(
      <ResourceTableItem {...defaultProps} resource={kubeItem} name={deployment.metadata.name} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("renders info about the resource status when given a list", () => {
    const deployment = {
      metadata: {
        name: "foo",
        selfLink: "/deployments/foo",
      },
      status: { replicas: 1, updatedReplicas: 1, availableReplicas: 1 },
    } as IResource;
    const kubeList = {
      isFetching: false,
      item: { items: [deployment] },
    };
    const wrapper = shallow(
      <ResourceTableItem
        {...defaultProps}
        resource={kubeList as any}
        name={deployment.metadata.name}
      />,
    );
    expect(wrapper.find(DeploymentItemRow)).toExist();
    expect(wrapper).toMatchSnapshot();
  });

  it("shows a warning when a list is empty", () => {
    const kubeList = {
      isFetching: false,
      item: { items: [] },
    };
    const wrapper = shallow(
      <ResourceTableItem {...defaultProps} resource={kubeList as any} name={""} />,
    );
    expect(wrapper.find("td")).toExist();
    expect(wrapper.find("td").text()).toEqual("No resource found");
  });

  it("skips an empty component if requested", () => {
    const kubeList = {
      isFetching: false,
      item: { items: [] },
    };
    const wrapper = shallow(
      <ResourceTableItem
        {...defaultProps}
        resource={kubeList as any}
        name={""}
        avoidEmptyResouce={true}
      />,
    );
    expect(wrapper.find("td")).not.toExist();
  });

  it("renders a ConfigMap", () => {
    const cm = {
      metadata: {
        name: "foo",
        selfLink: "/configmaps/foo",
      },
    } as IResource;
    kubeItem.item = cm;
    const wrapper = shallow(
      <ResourceTableItem {...defaultProps} resource={kubeItem} name={cm.metadata.name} />,
    );
    expect(wrapper.find(OtherResourceItem)).toExist();
  });

  it("it parses the type from the position in the self-link (it ignores the namespace)", () => {
    const deployment = {
      metadata: {
        name: "foo",
        selfLink: "/api/v1/namespaces/deployments/secrets/kubernetes",
      },
      status: { replicas: 1, updatedReplicas: 1, availableReplicas: 1 },
    } as IResource;
    kubeItem.item = deployment;
    const wrapper = shallow(
      <ResourceTableItem {...defaultProps} resource={kubeItem} name={deployment.metadata.name} />,
    );
    expect(wrapper.find(SecretItem)).toExist();
  });

  it("it parses the type from the position in the self-link (it ignores the resource name)", () => {
    const deployment = {
      metadata: {
        name: "foo",
        selfLink: "/api/v1/namespaces/default/secrets/services",
      },
      status: { replicas: 1, updatedReplicas: 1, availableReplicas: 1 },
    } as IResource;
    kubeItem.item = deployment;
    const wrapper = shallow(
      <ResourceTableItem {...defaultProps} resource={kubeItem} name={deployment.metadata.name} />,
    );
    expect(wrapper.find(SecretItem)).toExist();
  });
});
