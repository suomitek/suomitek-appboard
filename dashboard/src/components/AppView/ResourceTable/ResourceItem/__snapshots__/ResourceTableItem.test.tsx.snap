// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when fetching resources loading spinner matches the snapshot 1`] = `
<tr
  className="flex"
>
  <td
    className="col-3"
  >
    foo
  </td>
  <td
    className="col-9"
  >
    <LoadingWrapper
      loaded={false}
      type={1}
    />
  </td>
</tr>
`;

exports[`when fetching resources loading spinner matches the snapshot 2`] = `
<tr
  className="flex"
>
  <td
    className="col-3"
  >
    foo
  </td>
  <td
    className="col-9"
  >
    <LoadingWrapper
      loaded={false}
      type={1}
    />
  </td>
</tr>
`;

exports[`when there is a valid resouce renders info about the resource status 1`] = `
<tr
  className="flex"
>
  <DeploymentItemRow
    resource={
      Object {
        "metadata": Object {
          "name": "foo",
          "selfLink": "/deployments/foo",
        },
        "status": Object {
          "availableReplicas": 1,
          "replicas": 1,
          "updatedReplicas": 1,
        },
      }
    }
  />
</tr>
`;

exports[`when there is a valid resouce renders info about the resource status when given a list 1`] = `
<tr
  className="flex"
  key="/deployments/foo"
>
  <DeploymentItemRow
    resource={
      Object {
        "metadata": Object {
          "name": "foo",
          "selfLink": "/deployments/foo",
        },
        "status": Object {
          "availableReplicas": 1,
          "replicas": 1,
          "updatedReplicas": 1,
        },
      }
    }
  />
</tr>
`;
